diff -ruN nethack2/include/tradstdc.h nethack/include/tradstdc.h
--- nethack2/include/tradstdc.h	2024-11-06 17:39:01.701022880 -0300
+++ nethack/include/tradstdc.h	2024-11-06 17:39:01.757690010 -0300
@@ -432,7 +432,7 @@
 #if !defined(__linux__) || defined(GCC_URWARN)
 /* disable gcc's __attribute__((__warn_unused_result__)) since explicitly
    discarding the result by casting to (void) is not accepted as a 'use' */
-#define __warn_unused_result__ /*empty*/
+#define __warn_unused_result__ __unused__
 #define warn_unused_result /*empty*/
 #endif
 #endif
diff -ruN nethack2/include/unixconf.h nethack/include/unixconf.h
--- nethack2/include/unixconf.h	2024-11-06 17:39:01.701022880 -0300
+++ nethack/include/unixconf.h	2024-11-06 17:39:01.747689928 -0300
@@ -133,7 +133,7 @@
  * "extra output" method is used, but not all systems provide access to
  * a fine-grained timer.
  */
-/* #define TIMED_DELAY */ /* usleep() */
+#define TIMED_DELAY /* usleep() */
 #endif
 #if defined(MACOSX) && !defined(TIMED_DELAY)
 #define TIMED_DELAY
diff -ruN nethack2/sys/unix/hints/linux nethack/sys/unix/hints/linux
--- nethack2/sys/unix/hints/linux	2024-11-06 17:39:01.721023044 -0300
+++ nethack/sys/unix/hints/linux	2024-11-06 17:39:01.754356649 -0300
@@ -11,21 +11,21 @@
 
 #PREFIX=/usr
 PREFIX=$(wildcard ~)/nh/install
-HACKDIR=$(PREFIX)/games/lib/$(GAME)dir
-SHELLDIR = $(PREFIX)/games
+HACKDIR=$(PREFIX)/games/var/nethack
+SYSCONFDIR=/zip/var/nethack
+SHELLDIR = $(PREFIX)/usr/bin
 INSTDIR=$(HACKDIR)
-VARDIR = $(HACKDIR)
+VARDIR = games/var/nethack
 
 
 
-POSTINSTALL=cp -n sys/unix/sysconf $(INSTDIR)/sysconf; $(CHOWN) $(GAMEUID) $(INSTDIR)/sysconf; $(CHGRP) $(GAMEGRP) $(INSTDIR)/sysconf; chmod $(VARFILEPERM) $(INSTDIR)/sysconf;
+POSTINSTALL=cp -n sys/unix/sysconf $(SYSCONFDIR)/sysconf; $(CHOWN) $(GAMEUID) $(SYSCONFDIR)/sysconf; $(CHGRP) $(GAMEGRP) $(SYSCONFDIR)/sysconf; chmod $(VARFILEPERM) $(SYSCONFDIR)/sysconf;
 
-CFLAGS=-g -O -I../include -DNOTPARMDECL
+CFLAGS+=-Os -D_XOPEN_SOURCE_EXTENDED -D__COSMOPOLITAN__ -I../include -I/opt/cosmos/x86_64/include/ncurses -DNOTPARMDECL
 CFLAGS+=-DDLB
 CFLAGS+=-DCOMPRESS=\"/bin/gzip\" -DCOMPRESS_EXTENSION=\".gz\"
-CFLAGS+=-DSYSCF -DSYSCF_FILE=\"$(HACKDIR)/sysconf\" -DSECURE
-CFLAGS+=-DTIMED_DELAY
-CFLAGS+=-DHACKDIR=\"$(HACKDIR)\"
+CFLAGS+=-DSYSCF -DSYSCF_FILE=\"/zip/var/nethack/sysconf\" -DSECURE
+CFLAGS+=-DHACKDIR=\"nethack/games/var/nethack/\"
 CFLAGS+=-DDUMPLOG
 CFLAGS+=-DCONFIG_ERROR_SECURE=FALSE
 CFLAGS+=-DCURSES_GRAPHICS
@@ -37,7 +37,7 @@
 
 LINK=$(CC)
 # Only needed for GLIBC stack trace:
-LFLAGS=-rdynamic
+LFLAGS=$(LDFLAGS) -rdynamic
 
 WINSRC = $(WINTTYSRC) $(WINCURSESSRC)
 WINOBJ = $(WINTTYOBJ) $(WINCURSESOBJ)
@@ -52,6 +52,6 @@
 CHOWN=true
 CHGRP=true
 
-VARDIRPERM = 0755
-VARFILEPERM = 0600
-GAMEPERM = 0755
+VARDIRPERM = 0775
+VARFILEPERM = 0664
+GAMEPERM = 02755
diff -ruN nethack2/sys/unix/Makefile.top nethack/sys/unix/Makefile.top
--- nethack2/sys/unix/Makefile.top	2024-11-06 17:39:01.721023044 -0300
+++ nethack/sys/unix/Makefile.top	2024-11-06 17:39:01.761023371 -0300
@@ -22,16 +22,16 @@
 #PREFIX	 = /usr
 GAME     = nethack
 # GAME     = nethack.prg
-#GAMEUID  = games
-#GAMEGRP  = bin
+GAMEUID = root
+GAMEGRP = games
 
 # Permissions - some places use setgid instead of setuid, for instance
 # See also the option "SECURE" in include/config.h
 #GAMEPERM = 04755
-FILEPERM = 0644
+FILEPERM = 0664
 # VARFILEPERM = 0644
 EXEPERM  = 0755
-DIRPERM  = 0755
+DIRPERM  = 0775
 # VARDIRPERM = 0755
 
 # VARDIR may also appear in unixconf.h as "VAR_PLAYGROUND" else HACKDIR
